keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -30,9 +31,26 @@
     private final int TOP_LEVEL_INDENTATION = 1;
     private int indentation = TOP_LEVEL_INDENTATION;
 
+    private final int vertexCount;
+    private final Map<VertexId, Integer> vertexDegrees;
+    private final double averageVertexDegree;
+
     public BayesianNetwork(Set<? extends Vertex> vertices) {
         this.vertices = ImmutableList.copyOf(vertices);
         this.vertexLabels = buildLabelMap(vertices);
+
+        this.vertexCount = vertices.size();
+        this.vertexDegrees = calculateVertexDegrees(vertices);
```

Comment:
Is there any reason you're storing this for every Vertex?  It's not super difficult to calculate the degree for a particular Vertex (as in computationally expensive) so it seems a waste to store this even if people don't need it?

Even if we have something that could be expensive to calculate (like the average) - we could just cache that the first time someone uses it, instead of just calculating it in an unsolicited fashion?